### ISSUE TITLE
Web page not closing when option is selected

### ISSUE THREAD

**Issue by kolcs** [sdasda7777/ScheduleBlock#3]
I would like to block notifications about adblockplus updates. 
I made the following settings:
![image](https://github.com/sdasda7777/ScheduleBlock/assets/47258484/58c726a0-a8b7-42d6-a447-b32678660c61)

However when I open the following page: 
https://new.adblockplus.org/en/update?an=adblockpluschrome&ap=chrome&apv=120.0.0.0&av=3.21.1&p=chromium&pv=120.0.0.0

the tab keeps opened. I also tested it with facebook.com.

**Comment by sdasda7777** [comment]
Thanks, I think the issue might be the site is blocking scripts that don't have the same origin. Although you are able to enter the code, I cannot guarantee the browser will be able to execute it, iirc.

You might see reports about script origin violation in the developer console, I think.

**Comment by sdasda7777** [comment]
Oh, no, sorry for the mystification, the script execution probably has nothing to do with that. Are you sure that the regex matches the URL? You can copy the URL of the given page and test it in the tester on the settings page.

**Comment by kolcs** [comment]
Yes, I tested, with the regex checker and it was OK.
 Your first suggestion give me a hint, therefore I found a solution:
I redirect the page to an other one and close it. 
```
window.location = 'https://to/other/site';
window.close();
```

**Comment by kolcs** [comment]
Strange... It worked only once... Since then it only redirects me to any custom page I define but does not close the window...

**Comment by sdasda7777** [comment]
> Yes, I tested, with the regex checker and it was OK.

Yeah, you're right, it matches for me as well. However, for me it seems to work as expected, the window gets closed. Would you mind telling me what your browser, browser version and extension version are? 🤔

> Since then it only redirects me to any custom page I define but does not close the window...

That's actually the behaviour I would expect, yeah. I believe execution of scripts should be terminated when the page changes, to prevent malicious code stealing your cookies or whatever.

**Comment by kolcs** [comment]
Google Chrome
Version 120.0.6099.109 (Official Build) (64-bit)

**Comment by sdasda7777** [comment]
And what is the version of ScheduleBlock you are using?

**Comment by kolcs** [comment]
found it: 1.1.6

**Comment by sdasda7777** [comment]
Hmm, I'm not sure what the issue might be, this exact setup seems to be working on my machine. Are you using any extension that might be interfering? Could you try disabling all the other extensions? Also, could you show me the other rules you are using in case that is the issue?

**Comment by kolcs** [comment]
Settings:
![image](https://github.com/sdasda7777/ScheduleBlock/assets/47258484/063fed00-238b-4f76-9adc-aeabb87f9d75)

Disabled extensions:
![image](https://github.com/sdasda7777/ScheduleBlock/assets/47258484/a485b8fb-92f1-4877-aff5-7bcc9081abd0)

**Comment by sdasda7777** [comment]
I don't find anything suspicious about the rules. I take it it doesn't work even with the other extensions disabled? Also, could you try Facebook instead of `new.adblockplus.org` again?

**Comment by kolcs** [comment]
same...

**Comment by sdasda7777** [comment]
Anything of interest in the developer console? Unfortunately I can't do anything about this unless I can replicate it 😕

**Comment by kolcs** [comment]
I understand. I will try later on another computer...

**Comment by sdasda7777** [comment]
@kolcs Hi, did you manage to replicate it on a different machine?

**Comment by sdasda7777** [comment]
@kolcs Hi, do you have any new information for me to look into? (I will eventually close this issue unless that is the case)

**Comment by kolcs** [comment]
Sorry for the late answer. I tested it on different computers with my google profile. Then I even signed out and tested it with an empty profile and checked the console output:
![image](https://github.com/sdasda7777/ScheduleBlock/assets/47258484/3b3a50a1-d52e-4730-ab91-dd3ed26634e5)

It is the same regardless I am logged in to my profile or not.

**Comment by sdasda7777** [comment]
Hi, sorry for no reply. Does this happen to you every time you open the page? Does it happen regardless of how you open the page? I actually found one way to replicate this exact warning message, but it's not something I would personally stumble upon accidentally 😅

**Comment by sdasda7777** [comment]
From what I gather, the gist seems to be that for security reasons Chrome (and Firefox as well) doesn't allow closing of windows by scripts, unless in specific circumstances. For whatever reason for you the circumstances are never met, but it generally affects all users, as I said above. Unfortunately this is completely out of the extension's control, and achieving this functionality would mean circumventing Chrome's security, and then again every time the given vulnerability is fixed. Currently it seems I can only recommend using one of the other actions instead of closing the window, and perhaps chiming in on this request for change: https://issues.chromium.org/issues/40744259
